### PR TITLE
H-1488: Implement bulk-checking of permissions

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

When reading `N` entities at once we currently do `N` SpiceDB roundtrips. This should be reduced to ideally a single one.

## 🔍 What does this change?

- Bump SpiceDB to `1.28.0`
- Use the experimental bulk permission check endpoint

### 🚢 Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] are internal and do not require a docs change

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: libs/@local/hash-authorization/src/api.rs [hunk 1]
```diff
@@ -125,26 +125,7 @@ pub trait AuthorizationApi {
...
-                } = self
-                    .check_entity_permission(actor, permission, entity, consistency)
-                    .await?;
-                result.insert(entity, has_permission);
-                zookie = checked_at;
-            }
-            Ok((result, zookie))
-        }
-    }
+    ) -> impl Future<Output = Result<(HashMap<EntityUuid, bool>, Zookie<'static>), CheckError>> + Send;
```
Comment (indietyp) [inline libs/@local/hash-authorization/src/api.rs:128]: The API only mentions `EntityId` instead `EntityUuid`, except for this change. Why?

File: libs/@local/hash-authorization/src/backend/mod.rs [hunk 2]
```diff
@@ -382,7 +437,29 @@ impl CheckResponse {
     }
 }
 
-/// Error returned from [`ZanzibarBackend::check`].
+/// Return value for [`ZanzibarBackend::check_permissions`].
+#[derive(Debug)]
+#[must_use]
+pub struct BulkCheckResponse<I> {
+    /// If the subject has the specified permission or relation to an [`Resource`].
+    pub permission_iterator: I,
```
Comment (indietyp) [inline libs/@local/hash-authorization/src/backend/mod.rs:445]: nit: I dislike how this is suffixed with `_iterator`, in that case, it should just `I` should be constrained, that would maybe be a good idea anyway, considering `BulkCheckItem`. (or call it `permissions`)

non-blocking.

File: libs/@local/hash-authorization/src/zanzibar/api.rs [hunk 3]
```diff
@@ -152,10 +154,51 @@ where
...
+            .backend
+            .check_permissions(
+                entities
+                    .into_iter()
+                    .map(move |entity| (entity.entity_uuid, permission, actor)),
+                consistency,
+            )
+            .await?;
+        let mut status = Ok::<(), Report<RpcError>>(());
+        let permission_map = response
```
Comment (indietyp) [inline libs/@local/hash-authorization/src/zanzibar/api.rs:178]: nit: dislike the use of `_map` to indicate that it is indeed a map.

**Review by indietyp — CHANGES_REQUESTED** [review]
1 clarification
2 nits (non-blocking)

File: libs/@local/hash-authorization/src/api.rs (hunk 1 above)
Comment (TimDiekmann) [reply libs/@local/hash-authorization/src/api.rs:128]: SpiceDB only stores the UUIDs, so there is no way to get the owned-by-id back.

**Pull request merged**